src/sys/termios.rs

```diff
@@ -293,6 +359,82 @@ mod ffi {
             pub c_ospeed: speed_t
         }
 
+        #[derive(Clone, Copy, Debug, PartialEq)]
+        pub enum BaudRate {
+            B0,
+            B50,
+            B75,
+            B110,
+            B130,
+            B150,
+            B200,
+            B300,
+            B600,
+            B1200,
+            B1800,
+            B2400,
+            B4800,
+            B9600,
+            B19200,
+            B38400,
+            BOTHER,
+            B57600,
+            B115200,
+            B230400,
+            B460800,
+            B500000,
+            B576000,
+            B921600,
+            B1000000,
+            B1152000,
+            B1500000,
+            B2000000,
+            B2500000,
+            B3000000,
+            B3500000,
+            B4000000,
+        }
+
+        impl From<speed_t> for BaudRate {
+            fn from(s: speed_t) -> BaudRate {
+                match s {
+                    0 => BaudRate::B0,
+                    1 => BaudRate::B50,
+                    2 => BaudRate::B75,
+                    3 => BaudRate::B110,
+                    4 => BaudRate::B130,
+                    5 => BaudRate::B150,
+                    6 => BaudRate::B200,
+                    7 => BaudRate::B300,
+                    10 => BaudRate::B600,
+                    11 => BaudRate::B1200,
+                    12 => BaudRate::B1800,
+                    13 => BaudRate::B2400,
+                    14 => BaudRate::B4800,
+                    15 => BaudRate::B9600,
+                    16 => BaudRate::B19200,
+                    17 => BaudRate::B38400,
+                    10000 => BaudRate::BOTHER,
+                    10001 => BaudRate::B57600,
+                    10002 => BaudRate::B115200,
+                    10003 => BaudRate::B230400,
+                    10004 => BaudRate::B460800,
+                    10005 => BaudRate::B500000,
+                    10006 => BaudRate::B576000,
+                    10007 => BaudRate::B921600,
+                    10010 => BaudRate::B1000000,
+                    10011 => BaudRate::B1152000,
+                    10012 => BaudRate::B1500000,
+                    10013 => BaudRate::B2000000,
+                    10014 => BaudRate::B2500000,
+                    10015 => BaudRate::B3000000,
+                    10016 => BaudRate::B3500000,
+                    10017 => BaudRate::B4000000,
```

Comment:
I'm pretty sure this and many of the other constants you define here are completely incorrect as, looking at the linux kernel source (), the constants there are in octal.  These numbers are decimal and, as such, every constant you reference here greater than 7 is wrong.

Since there is better testing infrastructure for all libc variants in the `libc` crate, I believe we should introduce the constants there and add the type-safe enumeration providing access to this constants in nix.  That has been our basic approach.  Since `libc` uses `ctest`, it will verify the correctness of the constants.